### ISSUE TITLE
Tauren males and females lacking the 8 yard melee range they had in Vanilla (videos included)

### ISSUE THREAD

**Issue by ShadowzQueen** [LightsHope/server#726]
--

Write your report here

-The Patch is the current patch, Patch 1.7 on the Lightbringer server.

-The current issue is that while the Tauren's Hitbox has been increased just like Vanilla WoW, it is not working exactly as it used to. I was looking for information about this server, and a lot of people have been saying that the Tauren's Melee range is not actually 8 yards for their regular and special attacks (mortal strike, hamstring, etc) like in the previous Vanilla WoW. This was a known advantage that Tauren had, and even in Burning Crusade (when this would be fixed), people complained that Tauren could hit each other through the columns. Additionally I have heard that Tauren warriors on Light's Hope when using charge/intercept would be too far after using it to apply Hamstring immediately afterwards, which is not how this worked in Vanilla (see video #1 below)

-Expected behavior is that Tauren would have their melee range for regular attacks and melee attacks being 8 yards. This is because this was a double-edged sword, working such that other players can hit the Tauren hitbox in an 8 yard radius around the Tauren, but a Tauren's attacks that stated "melee range" would also hit at any distance 0-8 yards (see video #2 below). Things with fixed range like Whirlwind remained unchanged and functioned in the same range as other races.

-https://www.youtube.com/watch?v=RGBnjELkgok 2:28 shows Pat (Tauren Warrior) being close enough after charging to be able to use an attack immediately after, during the stun.

and

https://www.youtube.com/watch?v=t16CC47eI1g 0:50 shows the Tauren warrior being able to hit at a vast difference, male and female Taurens had the same range for their melee attacks (8 yards).

Here at arenajunkies they described the issue as it occured in the past with male and female tauren having this 8 yard melee range.

http://www.arenajunkies.com/topic/19398-tauren-swing-range-all/

That is everything, thank you for reading this.

**Comment by Cabro** [comment]
Hey, i have a bit of feedback to this. The yellow attacks have shorter range for ALL classes, my night elf warrior sometimes can autoattack but the yellow atacks (sunder/heroic strike etc) and "out of range".
I confirm this bug and make it general to all classes.

**Comment by Ravyd** [comment]
Being unable to use mortal strike or hamstring while autoing on my tauren warrior is extremely annoying. Now that you mention it, I have noticed being attacked while being unable to attack back. I thought it was just some sort of weird bug but this completely makes sense. Tauren are large and should have a large attack radius to compensate being hit from further away. If this bug becomes known, tauren could be at a massive disadvantage, having an effective dead zone against rogues, paladins and other warriors (along with mobs) from 5-8 yards like a hunter.

**Comment by Ravyd** [comment]
I can see NE's having a similar problem but not as bad as tauren. Not only are they tall they are wide. NE's are just tall. Similarly, if this scales gnomes could be at a massive advantage by being able to attack at 5 and not get attacked at >3-4 yrds

**Comment by ShadowzQueen** [comment]
This was programmed into the game, and was an obscure fact, and it didn't become prevalently known until BC due to Arena being on the cutting edge of balance. The issue I'm showing is specific to Tauren using melee, and only Tauren using melee (or being hit from farther away) for some reason. It was a disadvantage to non-warrior Tauren, while an interesting mechanic to Tauren. 

Here is a post on arena junkies in May 2008 (during BC) by a 3x gladiator, posting on his 3rd gladiator (his 2nd that he sold was a Tauren warrior).

"Every race have a swing range on autoattack and specials of 5 yards, except tauren which have a range of 8 yards on autoattack and specials. (the range on specials at one point was indicated on the abilities' tooltip, after a post on the warrior forums about this the tooltips were stripped of showing the actualy 'yard' range, and just indicate melee range now).

Think about everytime you've just been that little bit out of range to hit somebody, and think about how it would be if you had 3 extra yards. Things that would happen on my tauren:

*Getting an MS and autoattack mages that nova'd me and were 8 yards away.
*Rogues being unable to 6-8 kite
*Increased swing range = sweepingstrikes benefit
*If you're out of range of hitting the target (more then 8 yards) you can intercept. Where as with any other race, there is a distance at which you can neither hit the target or intercept (6-7 yards).

The drawback of being a tauren is that people can also hit you from 3 yards farther away compared to every other race, because you also have an increased sized hitbox. Another disadvantage is piercing howl and demo shout are fixed range on all races, and because of that you also have a smaller radius for them, when taking into account the larger hitbox taurens have. Whirlwind is only 5 yards for taurens. But the advantages of being a tauren outweigh the disadvantages here."

**Comment by Ravyd** [comment]
I totally agree this needs to be fixed. As it stands playing a tauren warrior is rediculous. The more I think about it, the more I can remember hitting an attack and having it redded out while I was autoing, and **being hit back.** Thought it was lag. It's not.

Charge works well on this server, but if it fails by even a little bit, that fact will be exponentially compounded by this problem in PVP.

**Comment by Ravyd** [comment]
Overall, I would say this is one of the most dire issues that needs attention at the moment. Tauren are probably 2nd or third most played horde side, meaning 1/4th+ of horde side players can be affected by this. Mainly warriors, hunters, and druids as well. I do not know if shapeshifted druids also have this problem at the moment, but if so, up to 1/3 of horde players are massively disadvantaged at the moment. Shamans to a much lesser degree, but enhancement shamans using stormstrike would be also affected. Making a hard life even harder

**Comment by Ravyd** [comment]
Cabro: as the OP stated, the tooltip says "melee range". Therefore if I can white hit you I can also hamstring you. Anything other than that is not working as intended

**Comment by Cabro** [comment]
Yes, the melee range is bugged.

**Comment by ShadowzQueen** [comment]
They don't know how to fix it.

We'll have to wait until Classic WoW for it to be fixed. = /

**Comment by GuybrushGit** [comment]
Side note: the Tauren hitbox has not been increased.

**Comment by SuddenDeath** [comment]
Here's how to fix it:

Taurens should have hitbox/combatreach of 4.05 yards.

Combat reach will change depending on if they are in a form that changes their model(druid forms, ghostwolf etc)

Proof:

Patch 2.4.2 video: https://www.youtube.com/watch?v=xjMRzrCOo2w
Source: http://www.warcraftmovies.com/movieview.php?id=82055

(The Tauren warr vs Tauren druid in the video obviously displays this quite a lot since they are both taurens it results in an almost double up situation, which explains the obnoxious long range)

https://youtu.be/UaHRC8PJkeE?t=2m50s
https://youtu.be/UaHRC8PJkeE?t=4m09s

https://youtu.be/oo_vcsdBKXc?t=12m
-doesn't get very close with Intercept(which is because of this too), yet still able to hit the druid right away


https://youtu.be/4JnQaXSHzEE?t=12m20s
https://youtu.be/oo_vcsdBKXc?t=7m50s


Some old forum threads about it:
http://www.dual-boxing.com/archive/index.php/t-7338.html (Also mentions cat form as tauren druid)
http://www.arenajunkies.com/topic/19398-tauren-swing-range-all/

Different hitbox in forms:
http://i.imgur.com/FDno7tU.gif (out of range when in travel form but not tauren)
Source: https://youtu.be/Rkb32GWDEzE?t=30m50s


How I determined approximately 4.0 value:
https://www.youtube.com/watch?v=geqf5cvuP7E
(9.18 - 1.33) / 2 = 3.925

Also if you check the hitbox of tauren NPC/Mobs ingame, they seem to have about 4.0 as well:
http://i.imgur.com/IxYc9Wr.jpg
https://youtu.be/5FvqqB_p8h4 - Still the same with tauren mobs on retail

Could probably be get the correct info with ".distance" command.

Old sniffs from retail TBC when a tauren druid shifted into caster form(credits to killerwife):

[0] (150) UNIT_FIELD_BOUNDINGRADIUS: 0.9747 [0] (151) UNIT_FIELD_COMBATREACH: 4.05 [0] (152) UNIT_FIELD_DISPLAYID: 59


ALSO
For this to work, there needs to be a proper melee range calculation in place.

How it's calculated in the client:
range_bonus = m_caster->GetCombatReach() + 1.33f + (target ? target->GetCombatReach() : m_caster->GetCombatReach());

Basically:
Melee range = max(5,combatreach_player + combatreach_target + 1.33)
